config.go

```diff
@@ -139,3 +118,20 @@ func setClientFromEnv(client *Client, log *zap.Logger) {
 		log.Warn("unknown exporter protocol", zap.String("env.name", exporterEnv), zap.String("env.value", exporterVal))
 	}
 }
+
+func fillValue(target *string, fromConf string, fromResource *resource.Resource, fromResourceKey attribute.Key, fromDefault string) {
```

Comment:
Not sure, I need to think. This is more like C/C++ style, rather than Go. I do like this style, but not sure if we need to bing it here. Give me a few days, I'll finish some urgent tickets and get back to this.

Reply:
I don't think Go has such an opinionated style, although its totally fair for your projects to pick a particular style